integrationTests/chainSimulator/staking/stakingProviderWithNodesinQueue_test.go

```diff
@@ -0,0 +1,130 @@
+	defer cs.Close()
+
+	mintValue := big.NewInt(0).Mul(big.NewInt(5000), oneEGLD)
+	validatorOwner, err := cs.GenerateAndMintWalletAddress(0, mintValue)
+	require.Nil(t, err)
+	require.Nil(t, err)
+
+	err = cs.GenerateBlocksUntilEpochIsReached(1)
+	require.Nil(t, err)
+
+	// create delegation contract
+	stakeValue, _ := big.NewInt(0).SetString("4250000000000000000000", 10)
+	dataField := "createNewDelegationContract@00@0ea1"
+	txStake := generateTransaction(validatorOwner.Bytes, getNonce(t, cs, validatorOwner), vm.DelegationManagerSCAddress, stakeValue, dataField, 80_000_000)
+	stakeTx, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(txStake, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, stakeTx)
+
+	delegationAddress := stakeTx.Logs.Events[2].Address
+	delegationAddressBytes, _ := cs.GetNodeHandler(0).GetCoreComponents().AddressPubKeyConverter().Decode(delegationAddress)
+
+	// add nodes in queue
+	_, blsKeys, err := chainSimulator.GenerateBlsPrivateKeys(1)
+	require.Nil(t, err)
+
+	txDataFieldAddNodes := fmt.Sprintf("addNodes@%s@%s", blsKeys[0], mockBLSSignature+"02")
+	ownerNonce := getNonce(t, cs, validatorOwner)
+	txAddNodes := generateTransaction(validatorOwner.Bytes, ownerNonce, delegationAddressBytes, big.NewInt(0), txDataFieldAddNodes, gasLimitForStakeOperation)
+	addNodesTx, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(txAddNodes, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, addNodesTx)
+
+	txDataFieldStakeNodes := fmt.Sprintf("stakeNodes@%s", blsKeys[0])
+	ownerNonce = getNonce(t, cs, validatorOwner)
+	txStakeNodes := generateTransaction(validatorOwner.Bytes, ownerNonce, delegationAddressBytes, big.NewInt(0), txDataFieldStakeNodes, gasLimitForStakeOperation)
+
+	stakeNodesTxs, err := cs.SendTxsAndGenerateBlocksTilAreExecuted([]*transaction.Transaction{txStakeNodes}, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.Equal(t, 1, len(stakeNodesTxs))
+
+	metachainNode := cs.GetNodeHandler(core.MetachainShardId)
+	decodedBLSKey0, _ := hex.DecodeString(blsKeys[0])
+	status := getBLSKeyStatus(t, metachainNode, decodedBLSKey0)
+	require.Equal(t, "queued", status)
+
+	// activate staking v4
+	err = cs.GenerateBlocksUntilEpochIsReached(int32(stakingV4ActivationEpoch))
+	require.Nil(t, err)
+
+	status = getBLSKeyStatus(t, metachainNode, decodedBLSKey0)
+	require.Equal(t, "unStaked", status)
+
+	ownerNonce = getNonce(t, cs, validatorOwner)
+	reStakeTxData := fmt.Sprintf("reStakeUnStakedNodes@%s", blsKeys[0])
+	reStakeNodes := generateTransaction(validatorOwner.Bytes, ownerNonce, delegationAddressBytes, big.NewInt(0), reStakeTxData, gasLimitForStakeOperation)
+	reStakeTx, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(reStakeNodes, maxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, reStakeTx)
+
+	status = getBLSKeyStatus(t, metachainNode, decodedBLSKey0)
+	require.Equal(t, "staked", status)
+
+	err = cs.GenerateBlocks(20)
```

Comment:
from linter: ineffectual assignment to err (ineffassign)

Reply:
fixed